src/spyglass/common/common_usage.py

```diff
@@ -408,3 +417,104 @@ def write_export(
         )
 
         # TODO: export conda env
+
+    def compile_dandiset(
+        self, key: dict, dandiset_id: str, dandi_api_key: str = None
+    ):
+        """Compile a Dandiset from the export.
+        Parameters
+        ----------
+        key : dict
+            ExportSelection key
+        dandiset_id : str
+            Dandiset ID generated by the user on the dadndi server
+        dandi_api_key : str, optional
+            API key for the dandi server. Optional if the environment variable
+            DANDI_API_KEY is set.
+        """
+        key = (self & key).fetch1("KEY")
+        paper_id = (self & key).fetch1("paper_id")
+        from .common_dandi import DandiPath, validate_dandiset
+
+        # make a temp dir with symbolic links to the export files
+        source_files = (self.File() & key).fetch("file_path")
+        paper_dir = f"{export_dir}/{paper_id}"
+        os.makedirs(paper_dir, exist_ok=True)
+        destination_dir = f"{paper_dir}/dandiset_{paper_id}"
+        os.makedirs(destination_dir, exist_ok=True)
+        for file in source_files:
+            if not os.path.exists(
+                f"{destination_dir}/{os.path.basename(file)}"
+            ):
+                os.symlink(file, f"{destination_dir}/{os.path.basename(file)}")
```

Comment:
I wrote the Export stuff above, in general, overwriting existing versions. What would happen if someone re-ran after upstream edits? There might be files in the dir whose names are outdated because they were generated with an incomplete set by current standards. I'd consider wiping the dandi directory whenever this is rerun, but I'm not sure what server-side operations that would require